### PR TITLE
[snapping] Fix broken version detection which leads to broken reading of snapping type flags

### PR DESCRIPTION
**Pull request by nirvn**
## Description

The regular expression used to extract the project file's version was wrong, and made it impossible to properly restore snapping type flags. This PR fixes that.

Were snapping flags adding in 3.12 or only master? Not sure this needs backporting.

### CODE REVIEW

**Comment by lbartoletti** [comment]
Thanks @nirvn 
It's only master.

**Comment by nirvn** [comment]
@lbartoletti , OK, noted. 

BTW, I've also noticed there's something wrong with the label in the advanced configuration widget's type column:
Screenshot from 2020-04-07 14-06-01
_Notice the absence of actual words in between the commas._

If you could look into that one, it'd be great.

File: src/core/qgssnappingconfig.cpp [hunk 1]
```diff
@@ -372,7 +372,7 @@ void QgsSnappingConfig::readProject( const QDomDocument &doc )
     if ( versionElem.hasAttribute( QStringLiteral( "version" ) ) )
     {
       version = versionElem.attribute( QStringLiteral( "version" ) );
-      QRegularExpression re( "(\\d).(\\d)" );
+      QRegularExpression re( "([\\d]*)\\.([\\d]*)" );
```
Comment (3nids) [inline src/core/qgssnappingconfig.cpp:375]:
```suggestion
      QRegularExpression re( "([\\d]+)\\.([\\d]+)" );
```
?

**Review by 3nids — APPROVED** [review]

**Comment by lbartoletti** [comment]
> If you could look into that one, it'd be great.

@nirvn I'll take a look, next week. Thanks

**Pull request merged**